### ISSUE TITLE
You choose MSVC or GCC convention?

### ISSUE THREAD

**Issue by iahunq** [LADSoft/OrangeC#826]
This issue is closed before I could comment: https://github.com/LADSoft/OrangeC/issues/821

My opinion is that either you follow the MSVC convention or the GCC convention. The MSVC convention is to use `.lib` for static or import libraries and `.obj` for object files. So far, I know Borland/Embarcadero, Digital Mars, OpenWatcom, and Pelles C all followed this convention. They are Windows-first compilers, that is, they were developed for Microsoft operating systems first, from DOS to Windows; porting to any other operating systems, if ever done, is a second thought. From this definition, Orange C is also a Windows-first compiler. I think it's reasonable to follow the MSVC convention like other Windows-first compilers. Or if you want to be more compatible with the open source compilers, you could follow the GCC convention, which uses `.a` for static or import libraries and `.o` for object files.

**Comment by GitMensch** [comment]
As noted in this issue, which is likely still open for anyone to comment on, the .a/.o are normally ELF, the .lib in MSVC format.
The convention should be that the filename extension matches the content, and OrangeC uses a different format (and also allows converting MSVC formats for import libraries), so it should stick with its own extension.

**Comment by iahunq** [comment]
> As noted in this issue, which is likely still open for anyone to comment on, the .a/.o are normally ELF

This is wrong. `.a` is only an `ar` archive (see: https://en.wikipedia.org/wiki/Ar_(Unix)). It can contain anything. `.o` could be PE/COFF too, at least with MinGW.

**Comment by GitMensch** [comment]
You are correct, normally != necessary.

Note that OrangeC's .l files _are_ a completely different format and changing file extensions of a "long-term application in use" is likely not a good idea.

**Comment by iahunq** [comment]
> You are correct, normally != necessary.
> 
> Note that OrangeC's .l files _are_ a completely different format and changing file extensions of a "long-term application in use" is likely not a good idea.

Then do you think the `.lib` produced by the Windows-first compilers I mentioned is the same or compatible? Nope. It's only a convention for file extensions. This is also another reason why I'm against following GCC convention to use `.a`. If you really made the conversion then you will be flooded with questions by people like whether they can use the `.a` import libraries created by GCC/Clang with Orange C. There are already many compilers use MSVC convention so people all know `.lib` is only a file extension and the `.lib` produced by different compilers are different and incompatible.

**Comment by LADSoft** [comment]
another problem though is eventually we would like to do this on linux, and I suspect we would get complaints using .lib there as well...

**Comment by iahunq** [comment]
> another problem though is eventually we would like to do this on linux, and I suspect we would get complaints using .lib there as well...

As far as I know, OpenWatcom is using the same MSVC convention they are using on Windows with their Linux version, and no one complained about that.
